documentation/source/factories.md

```diff
@@ -0,0 +1,207 @@
+        scales: [0.4, 1.6]
+```
+If you load the `.yaml` recipe as is into a python dictionary, you would get the following
+```python
+{
+  "train_dataset_params": {
+    "transforms": [
+      {
+        "SegColorJitter": {
+          "brightness": 0.1,
+          "contrast": 0.1,
+          "saturation": 0.1
+        }
+      },
+      {
+        "SegRandomFlip": {
+          "prob": 0.5
+        }
+      },
+      {
+        "SegRandomRescale": {
+          "scales": [0.4, 1.6]
+        }
+      }
+    ]
+  }
+}
+```
+
+This configuration alone is not very useful, as we need instances of the classes, not just their configurations.
+So we would like to somehow instantiate these classes `SegColorJitter`, `SegRandomFlip` and `SegRandomRescale`.
+
+Factories in SuperGradients come into play here! All these objects were registered beforehand in SuperGradients, 
+so that when you write these names in the recipe, SuperGradients will detect and instantiate them for you.
+
+## Registering a Class
+
+As explained above, only registered objects can be instantiated. 
+This registration consists of mapping the object name to the corresponding class type.
+
+In the example above, the string `"SegColorJitter"` was mapped to the class `SegColorJitter`, and this is how SuperGradients knows how to convert the string defined in the recipe, into an object.
+
+You can register the class using a name different from the actual class name. 
+However, it's generally recommended to use the same name for consistency and clarity.
+
+### Example
+
+```python
+from super_gradients.common.registry import register_transform
+
+@register_transform(name="MyTransformName")
+class MyTransform:
+    def __init__(self, prob: float):
+        ...
+```
+In this simple example, we register a new transform.
+Note that here we registered (for the sake of the example) the class `MyTransform` to the name `MyTransformName` which is different. 
+We strongly recommend to not do it, and to instead register a class with its own name.
+
+Once you registered a class, you can use it in your recipe. Here, we will add this transform to the original recipe
+```yaml
+train_dataset_params:
+  transforms:
+    - SegColorJitter:
+        brightness: 0.1
+        contrast: 0.1
+        saturation: 0.1
+    - SegRandomFlip:
+        prob: 0.5
+    - SegRandomRescale:
+        scales: [0.4, 1.6]
+    - MyTransformName:  # We use the name used to register, which may be different from the name of the class
+        prob: 0.7 
+```
+
+Final Step: Ensure that you import the module containing `MyTransformName` into your script. 
+Doing so will trigger the registration function, allowing SuperGradients to recognize it.
+
+Here is an example (adapted from the [train_from_recipe script](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/train_from_recipe.py)).
+
+```python
+from .my_module import MyTransform # Importing the module is enough as it will trigger the register_model function
```

Comment:
should be "register_transform" instead of "register_model"

Reply:
Fixed in [6b8e8ee](https://github.com/Deci-AI/super-gradients/pull/1395/commits/6b8e8ee8aee63068e1aee19cef7b314c06a44373)